inflate.c

```diff
@@ -946,8 +996,16 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
                 out -= left;
                 strm->total_out += out;
                 state->total += out;
-                if (INFLATE_NEED_CHECKSUM(strm) && (state->wrap & 4) && out)
-                    strm->adler = state->check = UPDATE(state->check, put - out, out);
```

Comment:
This block should be changed, so that "out" is checked first, as the two additions become no-op when out is 0.

Reply:
I'd wager that the two additions are less costly than an additional branch to the code and they make it less complicated.  The lines below could have maybe been used for that since out is checked before calling the inf_chksum routine but a lot of that is already bypass if on a platform (or rather, the platform) that doesn't need to compute checksums via our routines.

Reply:
It doesn't add another branch, just moves one before the additions... It also makes it possible to use LIKELY() or UNLIKELY() on just that branch for micro-optimization...